flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/common/CommonExecSink.java

```diff
@@ -174,6 +180,18 @@ protected Transformation<Object> createSinkTransformation(
                             inputParallelism));
         }
 
+        Object outputObject = null;
+        if (runtimeProvider instanceof OutputFormatProvider) {
```

Comment:
It's a bit strange that we support `OutputFormatProvider` & `SinkFunctionProvider` here, but not support `SinkProvider`. Since we've decided to drop support for deprecated sinks, it suffices to only support `SinkV2Provider`.

Reply:
I prefer to add the SinkProvider back to cover all of possible interfaces even through SinkProvider is marked as deprecated.